it/src/main/java/org/corfudb/universe/universe/docker/FakeDns.java

```diff
@@ -0,0 +1,187 @@
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+/**
+ * Fake DNS resolver which allows our tests to work well even though we use
+ * strange loopback IP addresses (127.x.y.z) with no corresponding reverse
+ * DNS.
+ * <p>
+ * This overrides the reverse lookups for such IPs to return the same address
+ * in String form.
+ * <p>
+ * Without this class, reverse DNS lookups for such addresses often take
+ * 5 seconds to return, causing timeouts and overall test slowness.
+ * <p>
+ * In the future this class might also be extended to test more interesting
+ * DNS-related scenarios.
+ */
+public class FakeDns {
+    private static FakeDns instance = new FakeDns();
+
+    private final Map<String, InetAddress> forwardResolutions = new HashMap<>();
+
+    private final Map<InetAddress, String> reverseResolutions = new HashMap<>();
+
+    /**
+     * whether the fake resolver has been installed
+     */
+    private boolean installed = false;
+
+    private FakeDns() {
+    }
+
+    public static FakeDns getInstance() {
+        return instance;
+    }
+
+    public synchronized void addForwardResolution(String hostname, InetAddress ip) {
+        forwardResolutions.put(hostname, ip);
+    }
+
+    public synchronized void addReverseResolution(InetAddress ip, String hostname) {
+        reverseResolutions.put(ip, hostname);
+    }
+
+    /**
+     * Install the fake DNS resolver into the Java runtime.
+     */
+    public synchronized FakeDns install() {
+        if (installed) return this;
+        try {
+            installDns();
+        } catch (Exception e) {
+            throw new IllegalStateException(e);
+        }
+        installed = true;
+
+        return this;
+    }
+
+    private void installDns() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException,
+            ClassNotFoundException, NoSuchFieldException {
+        try {
+            // Override the NameService in Java 9 or later.
+            final Class<?> nameServiceInterface = Class.forName("java.net.InetAddress$NameService");
+            Field field = InetAddress.class.getDeclaredField("nameService");
+            // Get the default NameService to fallback to.
+            Method method = InetAddress.class.getDeclaredMethod("createNameService");
+            method.setAccessible(true);
+            Object fallbackNameService = method.invoke(null);
+            // Create a proxy instance to set on the InetAddress field which will handle
+            // all NameService calls.
+            Object proxy = Proxy.newProxyInstance(
+                    nameServiceInterface.getClassLoader(),
+                    new Class<?>[]{nameServiceInterface},
+                    new NameServiceListener(fallbackNameService)
+            );
+            field.setAccessible(true);
+            field.set(InetAddress.class, proxy);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure that this accessibility bypass is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3011)